ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,264 @@
+# 8. HACBS Test Stream - API contracts
+
+Date: 2022-11-28
+
+## Status
+
+Accepted
+
+## Context
+
+The Hybrid Application Cloud Build Service being developed aims to serve Red Hat teams but also partners and customers. This requires a level of adaptability to avoid recreating custom flows and Tasks for each stakeholder.
+
+In this respect tasks developed by HACBS test stream should allow swapping external systems to accommodate different environments. This swap should not induce the complete recreation of pipelines.
+
+This and the idea of providing a homogeneous experience, which is easier to comprehend and navigate complex systems, leads to the definition of `API contracts`. These contracts need to be understood as guidance that may evolve with time and experience while keeping the aim of building a flexible homogeneous system.
+
+Tasks from the test stream may exchange information through
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is detailed results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+## Decision
+
+### Results and Records
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+  - The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+#### Tekton Results
+
+Each test will have a standardized short name in snake case, e.g. `release_label_required`, `architecture_label_deprecated` etc.
```

Comment:
Since we converted the failures to an integer field so it doesn't contain the test names in it anymore, I think we can move this line to the `Full Test Output JSON` section, probably somewhere around [line 135](https://github.com/redhat-appstudio/book/pull/24/files#diff-c480f27e8be91d3a13f9ed1429fff01b9790de544a57d7c812a25137a1898f61R135).

Reply:
Thanks, it is moved to the new [line 134](https://github.com/redhat-appstudio/book/pull/24/files#diff-c480f27e8be91d3a13f9ed1429fff01b9790de544a57d7c812a25137a1898f61R134)